hpc-software.md

```diff
@@ -0,0 +1,149 @@
+
+Information on how to access CHTC-managed installations is in the next 
+section of this guide. If you need to use a program not in that group, the instructions 
+for creating your own installation follow. 
+
+If you have questions or concerns about installing your own software or 
+the available dependencies, contact the facilitation team at chtc@cs.wisc.edu. 
+
+# 2. Using Pre-Installed Software in Modules
+
+All software on the cluster that is installed by CHTC staff is available via 
+a tool called "modules". 
+
+## A. See Available Software Modules
+
+There are two ways to search through the software modules on the HPC cluster: 
+
+1. **View all modules**
+	This command will show all software modules available: 
+		```
+		[alice@login]$ module avail
+		```
+
+2. **Search for specific modules**
+	If you are searching for a specific software module, you can use the 
+	`module spider` command with part of the software name. For example, to 
+	search for Open MPI modules, you would type:
+		```
+		[alice@login]$ module spider openmpi
+		```
+
+## B. Access Software in Modules
+
+Once you find a software module that you want to use, you need to "load" it 
+into your command line environment to make it active, filling in `module_name` with 
+the name you found through one of the above steps. 
+
+```
+[alice@login]$ module load module_name
+```
+
+> ** When to Load Modules **
+> 
+> You can load modules to compile code (see below). If you do this, make sure to load
+> the same modules as part of your job script before running the main command. 
+> 
+> You can also load modules to run specific software. If done for interactive 
+> testing, this should be done in an interactive job; otherwise, the module 
+> should be loaded in the job submit file. 
+
+## C. Unload Software in Modules
+
+If you no longer want to use a specific software installation, you can "unload"
+the software module with the following command: 
+
+```
+[alice@login]$ module unload module_name
+```
+
+If you want to clear your command line environment and start over, run the following: 
+
+```
+[alice@login]$ module purge
+```
+
+# 3. Installing Software on the Cluster
+
+## Overview
+
+Unless you are using a licensed software program provided via modules, you 
+are able to compile and install the software you need on the HPC Cluster. 
+Software should be installed to your `/software/username` 
+directory. If using CHTC's provided compilation tools via modules, make 
+sure to load the needed modules before compiling and to load the same 
+modules in your job submission. 
+
+For groups that would like to share software installations among group
+members, please contact us about getting a shared "group" directory. 
+
+If you are new to software installation, see the section below for 
+a more step-by-step description of the process. 
+
+## Step by Step Process
+
+1. **Download Source Code** - download the source code for your desired program to 
+	your `/software` directory.
```

Comment:
Recommending the source archive goes in `/software` is good for if the user wants to keep the source code around (good idea), but I'm wondering if it'll also lead to users leaving build cruft around in `/software` when they untar the source archive in the same directory and don't clean it up later.  Not the end of the world if (when) it happens, but a consideration.  Somewhere around here could be a good place to mention local scratch space.